src/test/rpc_wallet_tests.cpp

```diff
@@ -404,10 +404,7 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_z_exportwallet)
     BOOST_CHECK(addrs.size()==0);
 
     // wallet should have one key
-    auto address = pwalletMain->GenerateNewZKey();
-    BOOST_CHECK(IsValidPaymentAddress(address));
-    BOOST_ASSERT(boost::get<libzcash::SproutPaymentAddress>(&address) != nullptr);
-    auto addr = boost::get<libzcash::SproutPaymentAddress>(address);
+    libzcash::SproutPaymentAddress addr = pwalletMain->GenerateNewSproutZKey();
```

Comment:
Shouldn't we still have the assertions? The name change in this function does not ensure this check is done automatically for us.

Reply:
The function now returns the specific type rather than the variant so this is not necessary.